sspl_test/run_qa_test.sh

```diff
@@ -148,6 +148,15 @@ execute_test()
     $sudo $script_dir/run_sspl-ll_tests.sh $plan
 }
 
+if [ "$plan" == "self" ] || [ "$plan" == "self_primary" ]
+then
+    if [ "$IS_VIRTUAL" == "true" ]
+    then
+    echo "VM detected."
+    echo "ERROR: self or self_primary plan is intended to run on hardware setup."
```

Comment:
execution will still continue after printing the error. where we are halting the test?

Reply:
Added `exit 1` for that.